core-java-modules/core-java-collections-maps/src/test/java/com/baeldung/map/ImmutableMapUnitTest.java

```diff
@@ -1,84 +1,126 @@
-				.put("Costa Rica", "North America")
-				.build();
-		assertTrue(immutableMap.containsKey("USA"));
-		assertTrue(immutableMap.containsKey("Costa Rica"));
-		
-		assertThrows(UnsupportedOperationException.class, () -> immutableMap.put("Canada", "North America"));
-		
-		mutableMap.remove("USA");
-		assertTrue(immutableMap.containsKey("USA"));
-		
-		mutableMap.put("Mexico", "North America");
-		assertFalse(immutableMap.containsKey("Mexico"));
-	}
-	
-	@Test
-	@SuppressWarnings("deprecation")
-	public void whenGuavaImmutableMapFromOfMethod_thenOriginalCollectionChangesDoNotReflectInImmutableMap() {
-
-		ImmutableMap<String, String> immutableMap = ImmutableMap.of("USA", "North America", "Costa Rica", "North America");
-		assertTrue(immutableMap.containsKey("USA"));
-		assertTrue(immutableMap.containsKey("Costa Rica"));
-
-		assertThrows(UnsupportedOperationException.class, () -> immutableMap.put("Canada", "North America"));
-	}
-	
+import org.hamcrest.collection.IsMapContaining;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableMap;
+
+class ImmutableMapUnitTest {
+
+    @Test
+    void whenCollectionsUnModifiableMapMethod_thenOriginalCollectionChangesReflectInUnmodifiableMap() {
+
+        Map<String, String> mutableMap = new HashMap<>();
+        mutableMap.put("USA", "North America");
+
+        Map<String, String> unmodifiableMap = Collections.unmodifiableMap(mutableMap);
+        assertThrows(UnsupportedOperationException.class, () -> unmodifiableMap.put("Canada", "North America"));
+
+        mutableMap.remove("USA");
+        assertFalse(unmodifiableMap.containsKey("USA"));
+
+        mutableMap.put("Mexico", "North America");
+        assertTrue(unmodifiableMap.containsKey("Mexico"));
+    }
+
+    @Test
+    @SuppressWarnings("deprecation")
+    void whenGuavaImmutableMapFromCopyOfMethod_thenOriginalCollectionChangesDoNotReflectInImmutableMap() {
+
+        Map<String, String> mutableMap = new HashMap<>();
+        mutableMap.put("USA", "North America");
+
+        ImmutableMap<String, String> immutableMap = ImmutableMap.copyOf(mutableMap);
+        assertTrue(immutableMap.containsKey("USA"));
+
+        assertThrows(UnsupportedOperationException.class, () -> immutableMap.put("Canada", "North America"));
+
+        mutableMap.remove("USA");
+        assertTrue(immutableMap.containsKey("USA"));
+
+        mutableMap.put("Mexico", "North America");
+        assertFalse(immutableMap.containsKey("Mexico"));
+    }
+
+    @Test
+    @SuppressWarnings("deprecation")
+    void whenGuavaImmutableMapFromBuilderMethod_thenOriginalCollectionChangesDoNotReflectInImmutableMap() {
+
+        Map<String, String> mutableMap = new HashMap<>();
+        mutableMap.put("USA", "North America");
+
+        ImmutableMap<String, String> immutableMap = ImmutableMap.<String, String> builder()
+                .putAll(mutableMap)
+                .put("Costa Rica", "North America")
+                .build();
```

Comment:
When I format this file the space between the parameters and builder() is removed and the tab for the concatenated methods are fewer.

Reply:
Yes indeed :)
Done